src/sentry/monitors/models.py

```diff
@@ -591,3 +591,17 @@ def check_monitor_environment_limits(sender, instance, **kwargs):
         raise MonitorEnvironmentLimitsExceeded(
             f"You may not exceed {settings.MAX_ENVIRONMENTS_PER_MONITOR} environments per monitor"
         )
+
+
+@region_silo_only_model
+class MonitorEnvironment(Model):
+    __relocation_scope__ = RelocationScope.Excluded
+
+    monitor = FlexibleForeignKey("sentry.Monitor", related_name="incidents")
+    monitor_environment = FlexibleForeignKey("sentry.MonitorEnvironment", related_name="incidents")
+    opening_checkin = FlexibleForeignKey("sentry.MonitorCheckIn", related_name="created_incidents")
+    closing_checkin = FlexibleForeignKey(
```

Comment:
These feel like misnomers, I think we probably want 'starting_checkin` and `resolved_checkin`

and for good measure, we should probably add some comments to these fields to clearly indicate which checkins in the lifecycle of an incident these would be.

Also Important, these should probably be nullablabe **since the checkins will be removed**, in which case, I think we almost definitely would probably want to either

1. Also remove the incident when any checkins are removed
2. Keep start and end date and allow empty start and end check ins

Reply:
Makes sense